tests/preprocessing/test_label.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import sklearn.preprocessing as spp
+
+import dask.array as da
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from dask.array.utils import assert_eq as assert_eq_ar
+
+import dask_ml.preprocessing as dpp
+from dask_ml.utils import assert_estimator_equal
+
+
+choices = np.array(['a', 'b', 'c'], dtype=str)
+y = np.random.choice(choices, 100)
+y = da.from_array(y, chunks=13)
+s = dd.from_array(y)
+
+
+@pytest.fixture
+def pandas_series():
+    y = np.random.choice(['a', 'b', 'c'], 100)
+    return pd.Series(y)
+
+
+@pytest.fixture
+def dask_array(pandas_series):
+    return da.from_array(pandas_series, chunks=5)
+
+
+class TestLabelEncoder(object):
+    def test_basic(self):
+        a = dpp.LabelEncoder()
+        b = spp.LabelEncoder()
+
+        a.fit(y)
+        b.fit(y.compute())
+        assert_estimator_equal(a, b)
+
+    def test_input_types(self, dask_array, pandas_series):
+        a = dpp.LabelEncoder()
+        b = spp.LabelEncoder()
+
+        assert_estimator_equal(a.fit(dask_array),
+                               b.fit(pandas_series))
+
+        assert_estimator_equal(a.fit(pandas_series),
+                               b.fit(pandas_series))
+
+        assert_estimator_equal(a.fit(pandas_series.values),
+                               b.fit(pandas_series))
+
+        assert_estimator_equal(a.fit(dask_array),
+                               b.fit(pandas_series.values))
+
+    @pytest.mark.parametrize('array', [y, s])
+    def test_transform(self, array):
+        a = dpp.LabelEncoder()
+        b = spp.LabelEncoder()
+
+        a.fit(array)
+        b.fit(array.compute())
+
+        assert_eq_ar(a.transform(array).compute(),
+                     b.transform(array.compute()))
+
+    @pytest.mark.parametrize('array', [y, s])
+    def test_inverse_transform(self, array):
+
+        a = dpp.LabelEncoder()
+        assert_eq_ar(a.inverse_transform(a.fit_transform(array)).compute(),
+                     array.compute())
```

Comment:
Hopefully you don't need to call `.compute` here.  The `assert_eq` function should handle this for you.  It will also perform a number of other checks if it is provided dask objects directly.

Reply:
Currently the `.tranform` method is implemented such that `LabelEncoder.transform(<dask-series>)` returns a dask array. It looks like neither the `assert_eq` in `dask.array.utils` nor `dask.dataframe.utils` can compare a dask array and a dask series (although there could be a utility function I'm unaware of that will do this). 

One way around this could be to remove the `.compute`s and wrap dask series with `da.asarray`. This way `assert_eq` will get two arrays instead of an array and a series. Something like

```python
if isinstance(array, da.Array):
    assert_eq_ar(a.inverse_transform(a.fit_transform(array)), array)
if isinstance(array, dd.Series):
    assert_eq_ar(a.inverse_transform(a.fit_transform(array)), da.asarray(array))
```

Reply:
Does `dask_ml.utils._assert_eq` work for you?

Reply:
Actually, how about `assert_eq_ar(a.inverse_transform(a.fit_transform(array)), da.asarray(array))`